extension-manifest-v3/src/background/reporting.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+import AnonymousCommunication from '@whotracksme/webextension-packages/packages/anonymous-communication';
+import Reporting from '@whotracksme/webextension-packages/packages/reporting';
+
+import { observe } from '/store/options.js';
+
+function platformSpecificSettings() {
+  if (
+    /iPad|iPhone|iPod/.test(navigator.platform) ||
+    (navigator.platform === 'MacIntel' && navigator.maxTouchPoints > 1)
+  ) {
+    // Ghostery extension for Safari on iOS and other Apple mobile devices
+    return {
+      ALLOWED_COUNTRY_CODES: ['us', 'de', 'fr'],
+      PATTERNS_URL: 'https://cdn2.ghostery.com/wtm-safari-ios/patterns.json',
+      CHANNEL: 'safari-ios',
+    };
+  }
+
+  if (
+    /Safari/i.test(navigator.userAgent) &&
+    /Apple Computer/.test(navigator.vendor) &&
+    !/Mobi|Android/i.test(navigator.userAgent)
+  ) {
+    // Ghostery extension for Safari on MacOS (Desktop)
+    return {
+      ALLOWED_COUNTRY_CODES: ['us', 'de', 'fr'],
+      PATTERNS_URL:
+        'https://cdn2.ghostery.com/wtm-safari-desktop/patterns.json',
+      CHANNEL: 'safari-desktop',
+    };
+  }
+
+  console.warn(
+    'No matching config found. Falling back to patterns from Chrome Desktop.',
+  );
+  return {
+    ALLOWED_COUNTRY_CODES: ['us', 'de', 'fr'],
+    PATTERNS_URL: 'https://cdn2.ghostery.com/wtm-chrome-desktop/patterns.json',
+    CHANNEL: 'ghostery',
+  };
+}
+
+const COLLECTOR_DIRECT_URL = 'https://anonymous-communication.ghostery.net';
+const COLLECTOR_PROXY_URL = COLLECTOR_DIRECT_URL; // current we have no proxy configured
+
+const config = {
+  COLLECTOR_DIRECT_URL,
+  COLLECTOR_PROXY_URL,
+  CONFIG_URL: 'https://api.ghostery.net/api/v1/config',
+  ...platformSpecificSettings(),
+};
+console.debug('reporting config:', config);
```

Comment:
Clean up `console.debug` statement.

Reply:
I would prefer to log the settings. I didn't see any logger, so I used console.debug.

Reply:
I would rather make that code inside of the function (with those with `self` commented below), as this is useless for the user, and if used at all, only by us in the development process. Because of that, it can be only "uncomment", or run if required.

Otherwise, it only creates noise in the logs from the service worker.

Reply:
OK, I'll expose the config in "self" and remove the logging.